object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/CalledMethodsPredicateEvaluator.java

```diff
@@ -15,6 +29,118 @@ public CalledMethodsPredicateEvaluator(final Set<String> cmMethods) {
     this.cmMethods = cmMethods;
   }
 
+  /**
+   * Return true iff the boolean formula "lhs -> rhs" is valid, treating all variables as
+   * uninterpreted.
+   *
+   * <p>Used to determine subtyping between two @CalledMethodsPredicate annotations that are not
+   * identical.
+   */
+  public static boolean implies(final String lhsOrig, final String rhsOrig) {
+
+    JavaParser parser = new JavaParser();
+
+    String preamble = "class DUMMY { boolean DUMMY() { return ";
+    String afterward = "; } }";
+
+    String lhs = preamble + lhsOrig + afterward;
+    String rhs = preamble + rhsOrig + afterward;
+
+    CompilationUnit lhsAst = parser.parse(lhs).getResult().orElse(null);
+    CompilationUnit rhsAst = parser.parse(rhs).getResult().orElse(null);
+
+    if (lhsAst == null || rhsAst == null) {
+      return false;
```

Comment:
Returning `false` is surprising behavior and does not lead itself to a good debugging experience.  I think the checker should issue a warning so that the user knows there is a syntax error.  Otherwise, it could be very difficult to determine that there is a parenthesis missing, or `&` should have been `&&`, etc.

Reply:
The checker already issues an error for unparseable formulae in `ObjectConstructionVisitor`.

Reply:
(and there's no other way to create a predicate formula besides writing it by hand, or a conjunction of pure literals which is known to be parseable)

Reply:
Thanks for that information.  Can you note that in a comment?  In that case, I think it would be better to you change the `return false` to throwing an exception, to indicate that it cannot happen.

Reply:
I changed it to throw an informative exception